friends/src/components/EditFriend.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+class EditFriend extends Component {
+    constructor() {
+      super();
+
+      this.handleNameChange = this.handleNameChange.bind(this);
+      this.handleAgeChange = this.handleAgeChange.bind(this);
+      this.handleEmailChange = this.handleEmailChange.bind(this);
+
+      this.state = {
+          id: 0,
+          name: '',
+          age: 0,
+          email: ''
+      };
+    }
+
+    componentDidMount() {
+        this.setState({
+            id: this.props.currentFriend[0],
+            name: this.props.currentFriend[1],
+            age: this.props.currentFriend[2],
+            email: this.props.currentFriend[3],
+        });
+    }
+
```

Comment:
With this handler, you can deal with changes on all of your input fields with one handler instead of 3. 
```suggestion
handleInputChange = e => {
  this.setState({ [e.target.name]: e.target.value });
}
```